### PR TITLE
ControlledQubitUnitary should raise `DecompositionUndefinedError`

### PR DESCRIPTION
**Pull request by antalszava**
**Context**
By default, operations don't have a decomposition defined and raise `DecompositionUndefinedError` with the changes in https://github.com/PennyLaneAI/pennylane/pull/2024. Operations can, however, define there own decompositions.

`qml.QubitUnitary` has a custom decomposition defined. Its controlled version, `qml.ControlledQubitUnitary` inherits from `qml.QubitUnitary`, but doesn't have a decomposition defined. This, however, breaks examples because `qml.QubitUnitary`'s `compute_decomposition` is being called when a decomposition of `qml.ControlledQubitUnitary` is attempted (see related issue).

**Changes**

Explicitly defines the `qml.ControlledQubitUnitary.compute_decomposition` method as is in the `Operator` class to raise `DecompositionUndefinedError`.

**Related issue**
Closes https://github.com/PennyLaneAI/pennylane/issues/2219.

### CODE REVIEW

File: pennylane/ops/qubit/matrix_ops.py [hunk 1]
```diff
@@ -241,6 +241,10 @@ def __init__(
         total_wires = control_wires + wires
         super().__init__(*params, wires=total_wires, do_queue=do_queue)
 
+    @staticmethod
+    def compute_decomposition(*params, wires=None, **hyperparameters):
+        raise DecompositionUndefinedError
```
Comment (puzzleshark) [inline pennylane/ops/qubit/matrix_ops.py:246]: should we add control_wires as a parameter? or save that for later?

**Review by puzzleshark — APPROVED** [review]

File: pennylane/ops/qubit/matrix_ops.py (hunk 1 above)
Comment (antalszava) [reply pennylane/ops/qubit/matrix_ops.py:246]: @puzzleshark hmm the signature should accept that now too, right? Or we could make it more explicit to have it
```python
def compute_decomposition(U, control_wires, u_wires, control_values=None):
```

Comment (puzzleshark) [reply pennylane/ops/qubit/matrix_ops.py:246]: ya I was wondering if we should make it more explicit. good with either way though

Comment (antalszava) [reply pennylane/ops/qubit/matrix_ops.py:246]: Got it! Let's save it for later, perhaps we can provide some parts of the decomposition down the line and make the signature more explicit too.

**Comment by antalszava** [comment]
[sc-15739]

**Pull request merged**